### PR TITLE
chore(deps): bump go version to 1.23.1

### PR DESCRIPTION
**Pull request by rach-id**
## Description

Bumps go version to 1.23.1

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -29,16 +29,16 @@ jobs:
...
       - uses: actions/setup-go@v5
         with:
           go-version-file: 'go.mod'
 
-      - uses: golangci/golangci-lint-action@v4.0.0
+      - uses: golangci/golangci-lint-action@v6.1.0
         with:
           # Required: the version of golangci-lint is required and
           # must be specified without patch version: we always use the
           # latest patch version.
```
Comment (rootulp) [inline .github/workflows/lint.yml:41]: [optional] b/c this comment is outdated

```suggestion
```

**Review by rootulp — APPROVED** [review]
LGTM! golangci-lint bump looks like it surfaced some lint errors

**Review by rootulp — DISMISSED** [review]
[optional] instead of adding `/nolint:gosec` everywhere, we can disable that linter in .golangci.yml until we pull from upstream a copy of this codebase that does address those lint warnings

**Comment by rach-id** [comment]
> we can disable that linter in .golangci.yml until we pull from

I didn't want to do it so in case future contributions are hitting any cases where we should be aware. The current code worked for years without the checks, we can assume it's fine. But new code might need the safe conversions. That's why I left it

**Review by rootulp — APPROVED** [review]

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -12,25 +12,25 @@ jobs:
...
         with:
-          go-version: "1.22.6"
+          go-version: "1.23.1"
       - name: Create a file with all the pkgs
         run: go list ./... > pkgs.txt
       - name: Split pkgs into 4 files
         run: split -d -n l/4 pkgs.txt pkgs.txt.part.
       # cache multiple
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
```
Comment (staheri14) [inline .github/workflows/coverage.yml:21]: Just curious to know whether this is also necessary for the Go version upgrade to work? or it is an independent change?

Comment (rach-id) [reply .github/workflows/coverage.yml:21]: All are necessary because they don't support the new version of go, unless I am missing something

**Review by staheri14 — APPROVED** [review]

**Pull request merged**